### DISCUSSION TITLE
D1 Mini hardware setup

### DISCUSSION THREAD

**Discussion by bencarver** [syssi/esphome-seplos-bms#17] [Q&A] [answered]
Hi, this looks like a great project, and apologies for the dumb question.  How should the ESP8266/D1 Mini be connected to the Seplos RS485 port?  Do you directly connect TX to the RS485 A pin and RX to the RS485 B pin?  Or are you using a UART to RS485 converter such as an XY-017 (e.g. https://www.aliexpress.com/item/3256802054906940.html?gatewayAdapt=4itemAdapt).

Thanks for any guidance!

**Comment by syssi** [comment, marked as answer]
A RS485 converter is required! I've just added a description of the schematics: here:https://github.com/syssi/esphome-seplos-bms#schematics

Some Seplos models provides a RS232 port. This could also be used. In this case a RS232-to-TTL converter is required to shift the RS232 voltage level from -12/+12V to 0/3.3V.

**Reply by bencarver** [reply]
Excellent, thank you.  Here are draft PCB files including the gerber file (for PCB fabrication) as well as the schematic and PCB layout should anyone want to use or edit them.  I used EasyEDA software and jlcpcb.com to fabricate the PCBs.  These are designed for a D1 Mini, the XY-017 RS485 converter (https://www.aliexpress.com/item/3256802054906940.html) and the following RJ45 jack: https://www.aliexpress.com/item/2251832588241427.html.  I will report back if these work once I get the PCBs and parts.


[D1 Mini and RS485.zip](https://github.com/syssi/esphome-seplos-bms/files/8968977/D1.Mini.and.RS485.zip)
[Gerber_PCB_D1 mini and RS485 v2.1_2022-06-23.zip](https://github.com/syssi/esphome-seplos-bms/files/8968980/Gerber_PCB_D1.mini.and.RS485.v2.1_2022-06-23.zip)

**Reply by syssi** [reply]
Awesome! Do you like to share a screenshot? I must admit I'm not familiar with gerber files.

**Reply by bencarver** [reply]
Screenshot of the schematic:

![image](https://user-images.githubusercontent.com/5199151/175559312-1929c723-9309-441f-8ca7-d40f07d09b2c.png)


Screenshot of the PCB layout:

![image](https://user-images.githubusercontent.com/5199151/175559231-49bc16e6-ee5d-4de7-b76b-8d6a0811bd9b.png)

I'm certainly not too experienced with PCB design, so any feedback is welcome!

**Reply by bencarver** [reply]
I soldered the PCBs and components together as indicated in the posts above.  Everything works great.  I just needed to change the baud rate to 19200 for my particular Seplos BMS.  Now if anyone with 3D printing skills wants to make a case, we'd have a nice hardware solution!

![IMG_0139](https://user-images.githubusercontent.com/5199151/179844729-150163f6-dacf-40ba-ba6e-fe00d2074338.jpeg)

**Reply by syssi** [reply]
Good job!

**Reply by syssi** [reply]
Could you add/name a license for your material above? I would like to add the gerber files and images to the repository if possible. Feel free to contribute the files as pull request.

**Reply by bencarver** [reply]
Sorry for not responding to this. Feel free to add these files to the repo with whatever license you’re using in the repo. I’m not too good with pull requests or I’d do it myself (sorry!!).

**Reply by moonsorrox** [reply]
nice solution, very well done, is there any news
